### PR TITLE
Remember last opened screen on app startup (Nearby or Contributions)

### LINKED ISSUE #4808: Open the last opened screen of bottom navbar when launching the app
### What is the user problem or growth opportunity you want to see solved?

Whenever a users opens the app it starts in the "Contributions" screen. This might be useful for some users as they might prefer to be taken to the "Nearby" screen instead which they use often. OTOH, changing starting with "Nearby" screen by default might not help users who wish to see their Contributions first.

One possible solution that might satisfy both users is to open the last opened screen of the bottom navbar when launching the app. In this way, a user who leaves the app from the "Nearby" screen would see the "Nearby" screen when they open the app. Those who leave the app from the "Explore" screen would see it when they open the app. This would help the users to get to the screen they most use when that's one they were in when they leave the app.

**Note**: It is not possible now to leave the app from the "Nearby" / "Explore" screen by hitting back. Hitting back from those screens would take them to the "Contributions" screen from which they are able to exit the app. This needs to be changed in case we plan on implementing this.

### How do you know that this problem exists today? Why is this important?

Based on feedback from the users mailing list [[ref](https://groups.google.com/g/commons-app-android/c/nY9W4aBqAUw/m/5qWQcPxlEwAJ)]. It is important because it helps users get to their most used screen of the app quickly.

### Who will benefit from it?

Users who want screens other than the "Contributions" screen to be opened by default when the app opens.

### Anything else you would like to add?

_No response_

### PR DESCRIPTION
**Pull request by arinmodi**
Fixes #4808

**Tests performed**

Tested build variant on ASUS_X00TD with API level 28

### CODE REVIEW

**Comment by arinmodi** [comment]
@nicolas-raoul, please review this

**Comment by arinmodi** [comment]
@nicolas-raoul, review this, please

File: app/src/main/java/fr/free/nrw/commons/contributions/MainActivity.java [hunk 1]
```diff
@@ -147,6 +153,11 @@ private void setUpPager() {
                 // do not change title for more fragment
                 setTitle(item.getTitle());
             }
+            if(item.getTitle().equals(getString(R.string.nearby_fragment))){
```
Comment (4D17Y4) [inline app/src/main/java/fr/free/nrw/commons/contributions/MainActivity.java:156]: Is this the best way to do this ?

**Review by 4D17Y4 — CHANGES_REQUESTED** [review]
Looks good.
Also check for tests coverage for these conditions.

File: app/src/main/java/fr/free/nrw/commons/contributions/MainActivity.java (hunk 1 above)
Comment (arinmodi) [reply app/src/main/java/fr/free/nrw/commons/contributions/MainActivity.java:156]: I think Yes cause we need to look for only two screens when we open the app and if the user go to the the nearby screen and then close the app from that then and then we have to open nearby screen.

still is there any better way to do this ?

**Comment by arinmodi** [comment]
@4D17Y4, added the test and did some minor change, please review that, thanks

**Comment by arinmodi** [comment]
@nicolas-raoul, can you please review this

**Comment by arinmodi** [comment]
@4D17Y4, can you verify this ?

File: app/src/main/java/fr/free/nrw/commons/contributions/MainActivity.java [hunk 2]
```diff
@@ -130,8 +131,14 @@ public void onCreate(Bundle savedInstanceState) {
         } else {
             if(savedInstanceState == null){
                 //starting a fresh fragment.
-                setTitle(getString(R.string.contributions_fragment));
-                loadFragment(ContributionsFragment.newInstance(),false);
+                if(applicationKvStore.getBoolean("last_opened_nearby")){
+                    setTitle(getString(R.string.nearby_fragment));
```
Comment (4D17Y4) [inline app/src/main/java/fr/free/nrw/commons/contributions/MainActivity.java:135]: Can you add comments for these conditions.

File: app/src/main/java/fr/free/nrw/commons/contributions/MainActivity.java [hunk 3]
```diff
@@ -142,11 +149,13 @@ public void setSelectedItemId(int id) {
     }
 
     private void setUpPager() {
-        tabLayout.setOnNavigationItemSelectedListener(item -> {
+        tabLayout.setOnNavigationItemSelectedListener(navListener = (item) -> {
             if (!item.getTitle().equals(getString(R.string.more))) {
                 // do not change title for more fragment
                 setTitle(item.getTitle());
             }
+            applicationKvStore.putBoolean("last_opened_nearby",
```
Comment (4D17Y4) [inline app/src/main/java/fr/free/nrw/commons/contributions/MainActivity.java:157]: A brief comment statement on what we are doing here.

**Review by 4D17Y4 — CHANGES_REQUESTED** [review]
Tested it works well.
Code looks good. 
All tests passing. 

Just some minor changes cuz someone reading code should know why we are saving only nearby. 

Thanks.

**Comment by 4D17Y4** [comment]
Thanks for your patience @arinmodi.

**Comment by misaochan** [comment]
Just to clarify the functionality of this PR, this basically implements "Last opened screen (if it is Contributions or Nearby, otherwise Contributions)", correct?

**Comment by arinmodi** [comment]
> Just to clarify the functionality of this PR, this basically implements "Last opened screen (if it is Contributions or Nearby, otherwise Contributions)", correct?

@misaochan Absolutely, correct

**Review by misaochan — APPROVED** [review]

**Comment by misaochan** [comment]
LGTM, @4D17Y4 feel free to merge when your requested changes have been made. :)

**Comment by arinmodi** [comment]
@4D17Y4, comments added, please review thanks

**Review by 4D17Y4 — APPROVED** [review]
Looks good.

**Pull request merged**